playbooks/common/openshift-master/config.yml

```diff
@@ -310,12 +310,19 @@
     with_items: openshift_master_named_certificates
     when: named_certs_specified | bool
 
-- name: Configure master instances
+- name: Configure masters
   hosts: oo_masters_to_config
   any_errors_fatal: true
   serial: 1
   vars:
     sync_tmpdir: "{{ hostvars.localhost.g_master_mktemp.stdout }}"
+
+    # Do not pass a version if this is first master. (openshift_docker will sort it out)
+    # For subsequent masters we pass either pre-existing version for the master (if already installed),
+    # otherwise the first master version.
+    openshift_version: "{{ openshift.common.version if openshift.common.version is defined else oo_first_master.openshift.common.version.split('-')[0] }}"
+    when: inventory_hostname != groups.oo_first_master.0
```

Comment:
This sort of leads back to the need to duplicate the master config blocks. We need to setup first master without passing a version in, let it figure it out, then set_fact on other masters, then the config block for other masters.

I'm thinking now about just passing in a new var for the "primary_master_version" or something like this, then the subsequent masters can see if that's set, and use it if they don't have a version already.